examples/ContinuumSnakeWithLiftingWaveCase/continuum_snake_with_lifting_wave.py

```diff
@@ -0,0 +1,276 @@
+import pickle
+
+from elastica import *
+
+from continuum_snake_postprocessing import (
+    plot_snake_velocity,
+    plot_video,
+    compute_projected_velocity,
+    plot_curvature,
+)
+from snake_forcing import (
+    MuscleTorques_lifting,
+)
+from snake_contact import SnakeRodPlaneContact
+
+
+class SnakeSimulator(
+    BaseSystemCollection, Constraints, Forcing, Damping, CallBacks, Contact
+):
+    pass
+
+
+def run_snake(
+    b_coeff_lat,
+    PLOT_FIGURE=False,
+    SAVE_FIGURE=False,
+    SAVE_VIDEO=False,
+    SAVE_RESULTS=False,
+):
+    # Initialize the simulation class
+    snake_sim = SnakeSimulator()
+
+    # Simulation parameters
+    period = 2.0
+    final_time = 20.0
+    time_step = 5e-5
+    total_steps = int(final_time / time_step)
+    rendering_fps = 100
+    step_skip = int(1.0 / (rendering_fps * time_step))
+
+    # collection of snake characteristics
+    n_elem = 25
+    base_length = 0.35
+    base_radius = 0.009
+    snake_torque_ratio = 30.0
+    snake_torque_liftratio = 10.0
+
+    start = np.array([0.0, 0.0, 0.0 + base_radius])
+    direction = np.array([1.0, 0.0, 0.0])
+    normal = np.array([0.0, 0.0, 1.0])
+    density = 1000
+    E = 1e6
+    poisson_ratio = 0.5
+    shear_modulus = E / (poisson_ratio + 1.0)
+
+    shearable_rod = CosseratRod.straight_rod(
+        n_elem,
+        start,
+        direction,
+        normal,
+        base_length,
+        base_radius,
+        density,
+        youngs_modulus=E,
+        shear_modulus=shear_modulus,
+    )
+
+    snake_sim.append(shearable_rod)
+    damping_constant = 1e-1
+
+    # use linear damping with constant damping ratio
+    snake_sim.dampen(shearable_rod).using(
+        AnalyticalLinearDamper,
+        damping_constant=damping_constant,
+        time_step=time_step,
+    )
+
+    # Add gravitational forces
+    gravitational_acc = -9.80665
+
+    snake_sim.add_forcing_to(shearable_rod).using(
+        GravityForces, acc_gravity=np.array([0.0, 0.0, gravitational_acc])
+    )
+
+    # 1. Add muscle torques -- lateral wave
+    # Define lateral wave parameters
+    lateral_wave_length = 1.0
+    lateral_amp = b_coeff_lat[:-1]
+
+    lateral_ratio = 1.0  # switch of lateral wave
+    snake_sim.add_forcing_to(shearable_rod).using(
+        MuscleTorques,
+        base_length=base_length,
+        b_coeff=snake_torque_ratio * lateral_ratio * lateral_amp,
+        period=period,
+        wave_number=2.0 * np.pi / (lateral_wave_length),
+        phase_shift=0.0,
+        rest_lengths=shearable_rod.rest_lengths,
+        ramp_up_time=period,
+        direction=normal,
+        with_spline=True,
+    )
+
+    # 2. Add muscle torques -- lifting wave
+    # Define lifting wave parameters
+    lift_wave_length = lateral_wave_length
+    lift_amp = np.array([1e-3, 2e-3, 2e-3, 2e-3, 2e-3, 1e-3])
+
+    lift_ratio = 1.0  # switch of lifting wave
+    phase = 0.5
+    snake_sim.add_forcing_to(shearable_rod).using(
+        MuscleTorques_lifting,
```

Comment:
Change accordingly if the class name is changed

Reply:
changed